cmd/main.go

```diff
@@ -179,6 +179,21 @@ func parseConfig() (Config, error) {
 		retval.reconcile = *retval.ctrlConfig.Reconcile
 	}
 
+	if retval.ctrlConfig.EntryIDPrefix != "" && retval.ctrlConfig.EntryIDPrefixCleanup != nil && retval.ctrlConfig.EntryIDPrefix == *retval.ctrlConfig.EntryIDPrefixCleanup {
+		return retval, fmt.Errorf("if entryIDPrefixCleanup is specified, it can not be the same value as entryIDPrefix")
+	}
+
+	if retval.ctrlConfig.EntryIDPrefix != "" && retval.ctrlConfig.EntryIDPrefix[len(retval.ctrlConfig.EntryIDPrefix)-1] != "."[0] {
+		retval.ctrlConfig.EntryIDPrefix += "."
+	}
+	if retval.ctrlConfig.EntryIDPrefixCleanup != nil && *retval.ctrlConfig.EntryIDPrefixCleanup != "" && (*retval.ctrlConfig.EntryIDPrefixCleanup)[len(*retval.ctrlConfig.EntryIDPrefixCleanup)-1] != "."[0] {
```

Comment:
This is pretty unwieldy. I'd add a `addDotSuffix` function or something that took in a string and appended a `.` if-and-only-if the string is non-empty and doesn't already have the suffix. Then this could be simplified to:
```suggestion
	if retval.ctrlConfig.EntryIDPrefixCleanup != nil {
		*retval.ctrlConfig.EntryIDPrefixCleanup = addDotSuffix(*retval.ctrlConfig.EntryIDPrefixCleanup)
	}	
```

It could also be used unconditional above with EntryIDPrefix.